rerun-failed-run/action.yml

```diff
@@ -0,0 +1,54 @@
+---
+name: Rerun failed run
+
+description: Rerun a run which has failed
+
+inputs:
+  run-id:
+    description: The id of the run we wanna re-trigger.
+    required: true
+  repository:
+    description: The name of the repository in which the run exists.
+    required: true
+  error-message:
+    description: The error message which we want to test against
+    required: true
+  vault-addr:
+    description: The url of vault
+    required: false
+  vault-role-id:
+    description: The role id for vault
+    required: false
+  vault-secret-id:
+    description: The secret id for vault
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: Import Secrets
+      id: vault-secrets
+      uses: hashicorp/vault-action@affa6f04da5c2d55e6e115b7d1b044a6b1af8c74 # v2.7.4
```

Comment:
the latest version of `vault-action` is [3.0.0](https://github.com/hashicorp/vault-action/releases)
Also, why use a commit number instead of the semantic version?